### ISSUE TITLE
Provide more specific 'when' clause for default keybindings

### ISSUE THREAD

**Issue by alex-che** [microsoft/vscode-cpptools#662]
cpptools has the Alt+O keybinding for 'Switch Header/Source' command.
angular2-switcher has the Alt+O keybinding for 'Open Corresponding template(html) File'.
Having both these extensions makes only one keybinding to work.
Currently, cpptools has "editorTextFocus" for the "when" keybinding property, while angular2-switcher does not have the property set at all.
The following overrides in the user keybindings.json seems to fix this for me:

 ```
   {
        "key": "alt+o",
        "command": "extension.switchTemplate",
        "when": "editorTextFocus && editorLangId=='typescript'"
    },
    {
        "key": "alt+o",
        "command": "C_Cpp.SwitchHeaderSource",
        "when": "editorTextFocus && editorLangId=='cpp'"
    } 
```

Is it possible/reasonable to have this (or maybe some better) value for the "when" keybinding property out-of-the-box, to reduce the probability of keybindings conflicts with other extensions?

**Comment by bobbrow** [comment]
Thank you for reporting this.  We'll take a look.

**Comment by bobbrow** [comment]
This was fixed in version 0.11.1. Please reopen if you continue to have this issue.
